io-actions/src/fragment/collection.js

```diff
@@ -0,0 +1,77 @@
+const fetch = require('node-fetch');
+const { odinReferences } = require('./paths.js');
+
+function isCollection(body) {
+    const { path, fields } = body;
+    return path.indexOf('/collections/') > 0 && fields?.categories;
+}
+
+function parseReferences(ids, references) {
+    const cards = {};
+    const categories = [];
+
+    ids.forEach((id) => {
+        const reference = references[id];
+        const { fields } = reference?.value;
+        if (fields?.variant) {
+            //reference is a card
+            cards[id] = reference.value;
+        } else if (fields?.cards || fields?.categories) {
+            //reference is a category
+            const { label, cards, categories: categoryReferences } = fields;
+            if (cards?.length) {
+                //we consider categories with cards to be final
+                //(no subcategories)
+                categories.push({ label, cards });
+            } else {
+                const subReferences = parseReferences(
+                    categoryReferences,
+                    references,
+                );
+                categories.push({
+                    label,
+                    categories: subReferences.categories,
+                });
+            }
+        }
+    });
+
+    return { cards, categories };
+}
+
+/**
+ * attach categories to current collection fragment
+ */
+async function collection(context) {
+    const { body } = context;
+    const { id } = body;
+    if (isCollection(body)) {
+        const referencesPath = odinReferences(id, true);
+        try {
+            const response = await fetch(referencesPath);
+            if (response.status === 200) {
+                const root = await response.json();
+                const { references } = root;
+                const { cards, categories } = parseReferences(
```

Comment:
if this is card collection endpoint, let's call it card collection, if it is a generic collection api, maybe we should avoid hardcoding card logic?

Reply:
well for now it's card, i don't see other collections to be coming soon. Maybe we can make it generic when that makes sense?

Reply:
don't you want to call it card-collection api ?

Reply:
the API is still the same, it's fragment, i called it collection api but it's not, the content fragment model is called collection too, as for now collection is pretty clear for everyone. I can rename everything, but i'm not sure it's really clearer?

Reply:
renamed PR "collection transformer" :)